File: ADS-2-Semestre/EngenhariaDeSoftware/InterfacePatoC#/Pato/Pato.cs
using Pato;
using System;
using System.Collections.Generic;
using System.Text;

namespace JogoPato
{
    public abstract class Pato
    {
        protected IPadraoVoaveis comportamentoPato;

        protected IPadraoCorredor corridaPato;

        public abstract String mostrar();

        public String nadar()
        {
            return "Pato Nadando.";
        }

        public void setComportamento(IPadraoVoaveis padrao)
        {
            comportamentoPato = padrao;
        }

        public String ComportamentoPato()
        {
            if (comportamentoPato != null)
                return comportamentoPato.voar();

            if (corridaPato != null)
                return corridaPato.correndo();

            return "Pato sem comportamento";
        }

        public void setComportamentoCorrida(IPadraoCorredor padrao)
        {
            corridaPato = padrao;
        }
    }
}

File: ADS-2-Semestre/EngenhariaDeSoftware/InterfacePatoCorredor/Pato.java
package engsoft;

public abstract class Pato {
	
	protected Padrao_Voaveis comportamento_pato;
        
        protected Padrao_Corredor corrida_pato;

	abstract String mostrar();
	
	public String nadar()
	{
            return "Pato Nadando.";		
	}	
	
	public void setComportamento(Padrao_Voaveis padrao )
	{
            comportamento_pato = padrao;		
	}
	
	public String comportamento_pato()
	{		
            if (comportamento_pato != null)                        
                return comportamento_pato.voar();		

            if (corrida_pato != null) 
                return corrida_pato.correndo();

            return "Pato sem comportamento";
	}
        
        public void setComportamentoCorrida(Padrao_Corredor padrao )
	{
            corrida_pato = padrao;		
	}
	
	
}
